### PR TITLE
XRLeapProviderManager Automatic modes and explanations

### PR DESCRIPTION
**Pull request by MattGrayUL**
## Summary

- Adds a dropdown to swap between Automatic/OpenXR/LeapC to the XRLeapProviderManager
- Ensures all Prefabs are available via the GameObject context menu
- Adds an editor script to make the Inspector cleaner
- XRLeapProviderManager Automatic selects LeapC when in Editor or Windows to avoid OpenXR runtimes that don't present tracking 'quality' options yet

## Contributor Tasks

_These tasks are for the merge request creator to tick off when creating a merge request._

- [ ] Pair review with a member of the QA team.
- [ ] Add any release testing considerations to the MR for the next release.
- [x] Check any relevant CHANGELOG files have been updated.
- [x] Ensure documentation requirements are met e.g., public API is commented.
- [x] Consider any licensing/other legal implications for this MR e.g. notices required by any new libraries.
- [x] Add any relevant labels such as `breaking` to this MR.
- [ ] If this MR closes a Jira issue, make sure the fix version on the JIRA issue is set to the correct one.

## Reviewer Tasks

_Add any instructions or tasks for the reviewer such as specific test considerations before this can be merged._

[Use emojis in review threads to communicate intent and help contributors.](https://github.com/ultraleap/UnityPlugin/blob/develop/CONTRIBUTING.md#review-threads)

- [x] Code reviewed.
- [x] Non-code assets e.g. Unity assets/scenes reviewed.
- [x] Documentation has been reviewed. Includes checking documentation requirements are met and not missing e.g., public API is commented.
- [x] Checked and agree with release testing considerations added to MR for the next release.

## Closes JIRA Issue

Closes UNITY-941

### CODE REVIEW

File: Packages/Tracking/Core/Editor/Scripts/PrefabCreateMenu.cs [hunk 1]
```diff
@@ -14,19 +14,31 @@ namespace Leap
...
     {
-        [MenuItem("GameObject/Ultraleap/XR Leap Provider Manager")]
+        [MenuItem("GameObject/Ultraleap/XR/XR Leap Provider Manager", false, 0)]
         public static void CreateProviderXR()
         {
             CreatePrefab("XR Leap Provider Manager");
         }
 
-        [MenuItem("GameObject/Ultraleap/Service Provider (Desktop)")]
+        [MenuItem("GameObject/Ultraleap/XR/Service Provider (XR)", false, 11)]
```
Comment (MaxPalmer-UH) [inline Packages/Tracking/Core/Editor/Scripts/PrefabCreateMenu.cs:23]: How are you determining the priority here?

File: Packages/Tracking/Core/Runtime/Scripts/XRLeapProviderManager.cs [hunk 2]
```diff
@@ -62,16 +69,60 @@ private set
...
+        /// <param name="_source">A tracking source. Where possible, this should not be AUTOMATIC when this method is called</param>
+        void SelectTrackingSource(TrackingSource _source)
+        {
+            if(_source == TrackingSource.AUTOMATIC)
+            {
+                Debug.LogWarning("No specific Tracking Source selected. Automatically selecting Leap Service");
+                _source = TrackingSource.LEAP_SERVICE;
+            }
+            
+            if(_source == TrackingSource.OPEN_XR)
```
Comment (MaxPalmer-UH) [inline Packages/Tracking/Core/Runtime/Scripts/XRLeapProviderManager.cs:125]: Some weird spacing going on here. Does it need formatting?

File: Packages/Tracking/Core/Runtime/Scripts/XRLeapProviderManager.cs [hunk 3]
```diff
@@ -62,16 +69,60 @@ private set
         public override Frame CurrentFixedFrame => LeapProvider.CurrentFixedFrame;
 
         /// <summary>
-        /// An optional override to force the use of the LeapC tracking data
+        /// The chosen source of hand tracking data
         /// </summary>
-        [Tooltip("Forces the use of the non-OpenXR provider, using LeapC hand tracking")]
-        public bool forceLeapService;
+        [Tooltip("AUTOMATIC - Windows will use LeapService, Other platforms will use OpenXR if available and LeapService if OpenXR if not" +
```
Comment (MaxPalmer-UH) [inline Packages/Tracking/Core/Runtime/Scripts/XRLeapProviderManager.cs:74]: Other should not be captialized.

**Review by MaxPalmer-UH — COMMENTED** [review]
Looks fine code wise other than my comments. Will do some testing.

File: Packages/Tracking/Core/Editor/Scripts/PrefabCreateMenu.cs (hunk 1 above)
Comment (MattGrayUL) [reply Packages/Tracking/Core/Editor/Scripts/PrefabCreateMenu.cs:23]: I am using the MenuItem priority feature that provides separators here, as well as ordering the items in a specific way:

- XRLeapProviderManager
- OpenXR Provider
- XR Provider

MenuItems have a hidden feature where separator lines will be added if they are >10 'priority' apart, this priority allows it to look as follows 
image

File: Packages/Tracking/Core/Runtime/Scripts/XRLeapProviderManager.cs (hunk 2 above)
Comment (MattGrayUL) [reply Packages/Tracking/Core/Runtime/Scripts/XRLeapProviderManager.cs:125]: not sure I understand what you mean, but we will apply the auto-formatting patch at the point of release? :)

Comment (MattGrayUL) [reply Packages/Tracking/Core/Runtime/Scripts/XRLeapProviderManager.cs:125]: I will apply the formatting patch :)

**Review by MaxPalmer-UH — APPROVED** [review]
I'm happy with the code and Greg has reviewed this on the Pico 3 and HTC Vive Focus. I've tested it on a Valve Index.

**Pull request merged**